src/semantic_search/semantic_search/load_messages.py

```diff
@@ -164,54 +167,48 @@ def index_messages(channel_id, messages, start_from, pinecone_index, pinecone_na
     messages_for_embedding = list(filter(lambda emb_t: len(emb_t.text) != 0, messages_for_embedding))
     logging.info(f"Removed empty messages, {str(len(messages_for_embedding))} messages left")
 
-    insert_pinecone_embeddings(messages_for_embedding, pinecone_index, pinecone_namespace)
+    insert_db_embeddings(messages_for_embedding, postgres_cursor, namespace)
 
 
-def index_whole_channel(pinecone_namespace, channel_id):
+def index_whole_channel(namespace, channel_id):
     logging.info(f"Fetching all messages from {channel_id} channel")
-    messages = list(reversed(fetch_channel_messages(pinecone_namespace, channel_id)))
+    messages = list(reversed(fetch_channel_messages(namespace, channel_id)))
     logging.info(f"Loaded {str(len(messages))} messages")
 
     messages = filter_messages(messages)
     total_messages = len(messages)
     logging.info(f"Filtering out service messages, left {str(total_messages)} messages")
 
-    index_messages(channel_id, messages, 0, get_pinecone_index(), pinecone_namespace)
+    index_messages(channel_id, messages, 0, get_postgres_cursor(), namespace)
 
 
-def insert_pinecone_embeddings(messages_for_embedding: List[Embedding], pinecone_index, pinecone_namespace):
+def insert_db_embeddings(messages_for_embedding: List[Embedding], postgres_cursor, namespace):
     logging.info("Starting embeddings creation for the generated messages")
     chunk_size = 30  # for OpenAI
     embedding_chunks = [messages_for_embedding[i:i + chunk_size] for i in
                         range(0, len(messages_for_embedding), chunk_size)]
     counter = 0
     for chunk in embedding_chunks:
-        logging.info(f"Inserting a chunk of Pinecone embeddings: [{counter} - {counter + len(chunk) - 1}]")
+        logging.info(f"Inserting a chunk of Postgre embeddings: [{counter} - {counter + len(chunk) - 1}]")
         counter += len(chunk)
         try:
             embeddings = create_embeddings([embedding_message.text for embedding_message in chunk])
-            items = []
 
             for i in range(len(chunk)):
-                items.append({
-                    'id': chunk[i].id,
-                    'values': embeddings[i],
-                    'metadata': chunk[i].to_metadata()
-                })
-
-            pinecone_index.upsert(
-                vectors=items,
-                namespace=pinecone_namespace
-            )
+                metadata = json.dumps(chunk[i].to_metadata())
+                postgres_cursor.execute('INSERT INTO embedding (namespace, chunk_id, metadata, values) VALUES (%s, %s, %s, %s)', (namespace, chunk[i].id, metadata, embeddings[i]))
+            
+            postgres_commit()
```

Comment:
There is no way to commit chunks at once. We have to commit for the every chunks.

Reply:
Postgres does not support combined vector format(    
vectors=[
        (
         "vec1",                # Vector ID 
         [0.1, 0.2, 0.3, 0.4],  # Dense vector values
         {"genre": "drama"}     # Vector metadata
        ),
        (
         "vec2", 
         [0.2, 0.3, 0.4, 0.5], 
         {"genre": "action"}
        )
    ],) like pinecone. because Postgres support vector column like data type.